Classifier and Summarizer/text_media_matcher.py

```diff
@@ -0,0 +1,105 @@
+from sentence_transformers import SentenceTransformer
+from scipy.spatial.distance import cosine as cosine_distance
+
+class TextMediaMatcher:
+    ''' Class containing utilties to find the stable matching
+
+    Finds the stable matching by applying the Gale Shalpley algorithm
+    '''
+    def __init__(self,list_of_summary_sentences,list_of_media_descriptions):
+        self.list_of_summary_sentences = list_of_summary_sentences
+        self.list_of_media_descriptions = list_of_media_descriptions
+        self.set_size = len(list_of_media_descriptions) #size of each set 
+
+        self.sentence_embedding_model = SentenceTransformer('bert-base-nli-stsb-mean-tokens')
+        
+        # similarity matrix : will be used for sorting preferences 
+        # similarity between ith summary_sentence and jth media description
+        self.similarity_matrix = [[-1 for i in range(self.set_size)] for j in range(self.set_size) ] 
+
+        self.preference_of_sentences_for_media = [list(range(self.set_size)) for i in range(self.set_size)]
+        self.preference_of_media_for_sentences = [list(range(self.set_size)) for i in range(self.set_size)]
+    
+    def find_embeddings(self,list_of_sentences):
+        '''Returns the list of embeddings for a list of sentences'''
+        return self.sentence_embedding_model.encode(list_of_sentences)
+
+    def embed_text(self):
+        ''' embeds the two lists of summary sentences and image descriptions'''
+        self.embeddings_of_summary_sentences = self.find_embeddings(self.list_of_summary_sentences)
+        self.embeddings_of_media_descriptions = self.find_embeddings(self.list_of_media_descriptions)
+
+    def cosine_similarity(self,embedding_A,embedding_B):
+        '''returns the cosine similarity between two vectors'''
+        return 1.0-cosine_distance(embedding_A,embedding_B)
+
+    def form_preference_matrix(self):
+        ''' Builds and initializes the preference matrix for media and sentences'''
+        self.embed_text() # find the vector embeddings for both lists
+        for i in range(self.set_size):
+            for j in range(self.set_size):
+                #find the similarity between ith sentence and jth media
+                self.similarity_matrix[i][j] = self.cosine_similarity(self.embeddings_of_summary_sentences[i],self.embeddings_of_media_descriptions[j])
+        
+        for i in range(self.set_size):
+            self.preference_of_media_for_sentences[i].sort(key=lambda x:self.similarity_matrix[i][x],reverse=True)
+            self.preference_of_sentences_for_media[i].sort(key=lambda x:self.similarity_matrix[x][i],reverse=True)
+        
```

Comment:
The preference matrix uses only cosine similarity for scoring.
The scoring function mentioned in the doc will be implemented once the PR for the output of
the extractor is merged.